big_tests/tests/mod_http_upload_SUITE.erl

```diff
@@ -278,12 +276,49 @@ test_minio_upload(Config, ContentType) ->
             GetUrl = binary_to_list(extract_url(Result, <<"get">>)),
             PutUrl = binary_to_list(extract_url(Result, <<"put">>)),
             Header = generate_header(Config, ContentType),
-            PutRetValue = ibrowse:send_req(PutUrl, Header, put, ?MINIO_TEST_DATA),
-            ?assertMatch({ok, "200", _, []}, PutRetValue),
-            GetRetValue = ibrowse:send_req(GetUrl, [], get),
-            ?assertMatch({ok, "200", _, ?MINIO_TEST_DATA}, GetRetValue)
+            PutRetValue = request(put, PutUrl, Header, ?MINIO_TEST_DATA),
+            ?assertMatch({200, _}, PutRetValue),
+            GetRetValue = request(get, GetUrl, [], []),
+            ?assertMatch({200, ?MINIO_TEST_DATA}, GetRetValue)
         end).
 
+request(Method, PutUrl, Header, Data) ->
+    PURL = #{host := Host, port := Port, path := Path} = uri_string:parse(PutUrl),
+    {ok, ConnPid} = gun:open(Host, Port),
```

Comment:
Gun is not declared as a dependency of bigtests, and this only works because several of our dependencies declare gun as their dependency. That's not a good pattern for rebar, please add gun as a top-level dependency.
It's still a gain, to have one dependency less, whether it is declared or not. Eventually, we can also get rid of `shotgun` and simply use `gun`, eventually, gun 2.0. Eventually soon 😄

Reply:
to be honest, I thought that this is unnecessary in the framework of the current PR, due to the fact that this also partly concerns fusco etc. But done.

Reply:
If we're explicitly calling a library, that library should also be explicitly made a dependency, as a good release pattern 😕

Reply:
Agree, but I was mean - do it after update/remove/replace all deps, then collect info what deps is used in tests and then add them into rebar.config. Eg `fusco` is used, but not added, this info notaced and will be fixed soonish.